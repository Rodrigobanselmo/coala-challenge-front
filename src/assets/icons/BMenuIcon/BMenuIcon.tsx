import React, { FC } from "react";

import { IIconMuiProps } from "@/core/interfaces/IIconMuiProps";
import MenuIcon from "@mui/icons-material/Menu";

export const BMenuIcon: FC<IIconMuiProps> = ({ ...props }) => {
  return <MenuIcon {...props} />;
};

export default MenuIcon;
