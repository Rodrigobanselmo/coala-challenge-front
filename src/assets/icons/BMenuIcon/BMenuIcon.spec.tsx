import React from "react";
import { render, screen } from "@testing-library/react";
import { BMenuIcon } from "./BMenuIcon";

describe("BMenuIcon", () => {
  it("renders without errors", () => {
    render(<BMenuIcon />);
    const icon = screen.getByTestId("MenuIcon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("applies custom props", () => {
    render(<BMenuIcon fontSize="large" color="primary" />);

    const icon = screen.getByTestId("MenuIcon");
    expect(icon).toMatchSnapshot();
  });
});
