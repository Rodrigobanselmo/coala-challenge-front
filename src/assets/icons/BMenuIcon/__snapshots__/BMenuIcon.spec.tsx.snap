// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BMenuIcon applies custom props 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
  data-testid="MenuIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
  />
</svg>
`;

exports[`BMenuIcon renders without errors 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="MenuIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
  />
</svg>
`;
